requests_kerberos/kerberos_.py

```diff
@@ -108,8 +111,14 @@ def generate_request_header(self, response, host, is_preemptive=False):
 
         try:
             kerb_stage = "authGSSClientInit()"
+            # contexts still need to be stored by host, but realm_override
+            # allows use of an arbitrary kerberos realm (eg, in cases of
+            # aliased hosts, internal vs external, CNAMEs w/ name-based
+            # HTTP hosting)
             result, self.context[host] = kerberos.authGSSClientInit(
-                "{0}@{1}".format(self.service, host), gssflags=gssflags)
+                "{0}@{1}".format(self.service,
+                     host if not self.realm_override else self.realm_override),
+                gssflags=gssflags, principal=self.principal)
```

Comment:
This is getting really dense. Can we pull the string formatting out of the call to `authGSSClientInit` so that we can more clearly see what's going on?